pg_search/src/postgres/expression.rs

```diff
@@ -0,0 +1,49 @@
+use pgrx::{
+    pg_sys::{self, AsPgCStr},
+    PgList, PgRelation,
+};
+
+// find_funcexpr_idx returns the attribute number of the
+// node in the index.
+pub fn find_funcexpr_attnum(indexrel: &PgRelation, node: *mut pg_sys::Node) -> Option<i32> {
+    let index_info = unsafe { *pg_sys::BuildIndexInfo(indexrel.as_ptr()) };
+    let heaprel = indexrel
+        .heap_relation()
+        .expect("index relation should have a heap relation");
```

Comment:
The caller has just finished finding the indexrel for this heaprel, so you should be able to pass the heaprel as well.